classes/Directory.js

```diff
@@ -36,20 +38,56 @@ class Directory {
         }
       })
   
-      if (resp.status !== 200) return {}
-  
-      const directories = resp.data.map(object => {
-        const pathNameSplit = object.path.split("/")
-        const dirName = pathNameSplit[pathNameSplit.length - 1]
-        if (object.type === 'dir') {
-          return {
-            path: encodeURIComponent(object.path),
-            dirName
-          }
+      if (resp.status !== 200) {
+        if (this.dirType instanceof FolderType) throw new BadRequestError(`Path ${this.dirType.getFolderName()} was invalid!`)
+        return {}
+      }
+
+      const filesOrDirs = resp.data.map((fileOrDir) => {
+        const {
+          name,
+          path,
+          sha,
+          size,
+          content,
+          type,
+        } = fileOrDir
+        return {
+          name,
+          path,
+          sha,
+          size,
+          content,
+          type,
         }
```

Comment:
This should occur inside the `FolderType` check after the validation

Reply:
Thanks for pointing this out - fixed with [83f314d](https://github.com/isomerpages/isomercms-backend/pull/108/commits/c6f99d1b32006316f97f364694e2e622d3222bea)